### PR TITLE
Simplify javabin path resolution

### PR DESCRIPTION
**Pull request by Bencodes**
https://github.com/bazelbuild/rules_android/issues/151

### CODE REVIEW

**Comment by jylinv0** [comment]
Hi Bencodes,
I added tests for android_local_test, could you rerun the checks?

**Comment by Bencodes** [comment]
@jylinv0 I rebased this branch so the tests can run again.

It looks like the `android_local_test` tests are failing on the `main` branch right now and might need to be fixed before this PR can be merged https://buildkite.com/bazel/rules-android/builds/1935#018b05b8-e1b6-455f-9a98-c6f31693f0d1

**Comment by Bencodes** [comment]
@jylinv0 thanks for fixing the tests in `main`! This branch is rebased against everything and should be good to review.

File: rules/android_local_test/impl.bzl [hunk 1]
```diff
@@ -255,21 +255,11 @@ def _process_deploy_jar(ctx, java_package, jvm_ctx, proto_ctx, resources_ctx, **
...
     javabase = ctx.attr._current_java_runtime[java_common.JavaRuntimeInfo]
-    java_executable = str(javabase.java_executable_runfiles_path)
+    java_executable = javabase.java_executable_exec_path
     java_executable_files = javabase.files
 
-    # Absolute java_executable does not require any munging
-    if java_executable.startswith("/"):
-        java_executable = "JAVABIN=" + java_executable
-
-    prefix = ctx.attr._runfiles_root_prefix[BuildSettingInfo].value
```
Comment (Bencodes) [inline rules/android_local_test/impl.bzl:265]: I can remove `_runfiles_root_prefix` as well but I suspect it will break internal tests if it's being used.

Comment (jylinv0) [reply rules/android_local_test/impl.bzl:265]: I'll take care of it.

**Pull request merged**